src/utils/datetime/validator.test.ts

```diff
@@ -0,0 +1,52 @@
+import {isValid} from './validator'
+
+describe('validation tests', () => {
+  it('should do the thing', function() {
+    expect(isValid('1999-02-09 23:00:00', 'YYYY-MM-DD HH:mm:ss')).toBeTruthy()
+    expect(
+      isValid('1999-02-09 12:00:00 PM -0700', 'YYYY-MM-DD hh:mm:ss a ZZ')
```

Comment:
Here, we are testing for local timezone, but it's okay because we provide the timezone. There will be no conflict in the CI.